### PR TITLE
add ban/unban user

### PR DESCRIPTION
**Pull request by diPhantxm**
- add ban/unban endpoints
- unit tests

### CODE REVIEW

File: moderation.go [hunk 1]
```diff
@@ -42,3 +42,68 @@ func (c *Client) GetBannedUsers(params *BannedUsersParams) (*BannedUsersResponse
...
+}
+
+type ManyBanUser struct {
+	Bans []BanUser `json:"data"`
+}
+
+type BanUser struct {
+	BoardcasterId string `json:"broadcaster_id"`
+	CreatedAt     string `json:"created_at"`
+	EndTime       string `json:"end_time"`
```
Comment (nicklaw5) [inline moderation.go:70]: Both `CreatedAt` and `EndTime` fields should be of type [Time](https://github.com/nicklaw5/helix/blob/main/time.go#L17-L19). This allows the our custom [UnmarshalJSON](https://github.com/nicklaw5/helix/blob/main/time.go#L26-L37) to handle converting the string represented time to a native [time.Time](https://pkg.go.dev/time#Time) type.

See the [User type](https://github.com/nicklaw5/helix/blob/main/users.go#L16) as an example.

File: moderation_test.go [hunk 2]
```diff
@@ -92,3 +93,187 @@ func TestGetBannedUsers(t *testing.T) {
...
+			}
+
+			if resp.Data.Bans[0].UserId != testCase.params.Body.UserId {
+				t.Errorf("expected user id to be \"%s\", got \"%s\"", testCase.params.Body.UserId, resp.Data.Bans[0].UserId)
+			}
+
+			if resp.Data.Bans[0].EndTime != "" {
+				layout := "2006-01-02T15:04:05Z"
+				createdTime, _ := time.Parse(layout, resp.Data.Bans[0].CreatedAt)
+				endTime, _ := time.Parse(layout, resp.Data.Bans[0].EndTime)
```
Comment (nicklaw5) [inline moderation_test.go:179]: By actioning the above comment, you won't need to perform `time.Parse` here. The parsing occur when the JSON unmarshaling happens. These three lines can be removed.

File: moderation_test.go [hunk 3]
```diff
@@ -92,3 +93,187 @@ func TestGetBannedUsers(t *testing.T) {
...
+
+				expireTime := createdTime.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))
+
+				if !expireTime.Equal(endTime) {
+					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
+				}
```
Comment (nicklaw5) [inline moderation_test.go:185]: These lines can then be written like so:

```suggestion
				expireTime := resp.Data.Bans[0].CreatedAt.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))

				if !expireTime.Equal(resp.Data.Bans[0].EndTime) {
					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
				}
```

**Review by nicklaw5 — CHANGES_REQUESTED** [review]
Thanks for contributing. Looks good. We just need to adjust the time field types. See the comments I've left.

**Comment by diPhantxm** [comment]
Thanks for your comment. I've fixed data types as you said. Also added comment to unban user function and noticed one more defect in test function.
Didn't notice your data type for time right away. It's actually my first time contributing to an open source project.

**Comment by nicklaw5** [comment]
Thanks and welcome to open source development :slightly_smiling_face:

**Pull request merged**